NSS-Day-Cohort-71/nss-day-cohort-71-bangazon-api-bangazon-api
# Add Backend Support for Viewing Past Orders with Total and Payment Method

Description of PR that completes issue here...

## Changes

- Added support in the backend for retrieving past orders with details including order date, total, and payment method.
- Implemented a method to calculate the total for each order by summing the prices of the associated line items.
- Obscured the payment method to display only the last four digits of the payment account number.
- Added handling for cases where no payment method is associated with an order.

---

## Requests / Responses

If this PR contains code that defines a new request/response, or changes an existing one, please put the JSON representations here.

**Request**

GET `/orders` Retrieves the list of past orders for the authenticated user.

```json
{
    "title": "Kite",
    "product_type_id": 1,
    "description": "Red. It flies high.",
    "quantity": 5
}
```

**Response**

HTTP/1.1 201 OK

**Example Request**:

```json
{
  "Authorization": "Token your-token-here"
}
Example Response:

HTTP/1.1 200 OK

json
Copy code
[
  {
    "id": 1,
    "created_date": "2019-08-16",
    "total": 0.00,
    "payment_type": "**** **** **** 8439"
  },
  {
    "id": 2,
    "created_date": "2019-03-26",
    "total": 2962.33,
    "payment_type": "**** **** **** 8439"
  },
  {
    "id": 3,
    "created_date": "2019-07-01",
    "total": 0.00,
    "payment_type": "**** **** **** 8439"
  },
  {
    "id": 4,
    "created_date": "2019-03-14",
    "total": 1655.15,
    "payment_type": null
  }
]
Error Response (if token is invalid)
HTTP/1.1 401 Unauthorized

json
Copy code
{
  "detail": "Invalid token."
}
Example Response:

HTTP/1.1 200 OK

json
Copy code
[
  {
    "id": 1,
    "created_date": "2019-08-16",
    "total": 0.00,
    "payment_type": "**** **** **** 8439"
  },
  {
    "id": 2,
    "created_date": "2019-03-26",
    "total": 2962.33,
    "payment_type": "**** **** **** 8439"
  },
  {
    "id": 3,
    "created_date": "2019-07-01",
    "total": 0.00,
    "payment_type": "**** **** **** 8439"
  },
  {
    "id": 4,
    "created_date": "2019-03-14",
    "total": 1655.15,
    "payment_type": null
  }
]
Error Response (if token is invalid)
HTTP/1.1 401 Unauthorized

json
Copy code
{
  "detail": "Invalid token."
}
```

## Testing

Description of how to test code...

- [ ] Seed database

1. Run migrations to ensure the database schema is up to date.
2. Seed the database with some test orders and payments for authenticated users.
3. Use the /orders endpoint with an authorized request to retrieve past orders.
4. Verify the following fields in the response:

- Order ID
- Created date
- Total (displaying $0.00 for orders with no line items or totals)
- Payment method (obscured with the last four digits or "No payment method" for orders without one).


## Related Issues

- Fixes #35
